docs/experiments-msmarco-irst.md

```diff
@@ -0,0 +1,226 @@
+# Pyserini: IRST on MS MARCO Passage and Document
+
+This page describes how to reproduce IRST experiments with the IBM model on the MS MARCO collections.
+
+
+## Passage Reranking 
+
+### Data Preprocessing
+
+For IRST, we make the corpus as well as the pre-built indexes available to download.
+
+> You can skip the data prep and indexing steps if you use our pre-built indexes. Skip directly down to next section below.
+
+Here, we start from MS MARCO [passage corpus](https://github.com/castorini/pyserini/blob/master/docs/experiments-ltr-msmarco-passage-reranking.md) that has already been processed.
+As an alternative, we also make available pre-built indexes (in which case the indexing step can be skipped).
+
+
+We run the below commands to obtain pre-built index in cache.
+
+```bash
+python -c "from pyserini.search.lucene import LuceneSearcher; LuceneSearcher.from_prebuilt_index('msmarco-passage-ltr')"
+```
+
+To confirm, `index-msmarco-passage-ltr-20210519-e25e33f.tar.gz` has an MD5 checksum of `a5de642c268ac1ed5892c069bdc29ae3`.
+
+The below scripts convert queries to json objects with text, text_unlemm, raw, and text_bert_tok fields
+
+
+```bash
+mkdir irst_test
+python scripts/ltr_msmarco/convert_queries.py --input path_to_topics --output irst_test/queries.irst_topics.dev.small.json
+```
+
+We can now perform retrieval to gain the baseline:
+
+```
+python -m pyserini.search.lucene \
+  --index ~/.cache/pyserini/indexes/index-msmarco-passage-ltr-20210519-e25e33f.a5de642c268ac1ed5892c069bdc29ae3/ \
+  --topics topics \
+  --output runs/run.topics.bm25tuned.txt \
+  --output-format trec \
+  --hits 1000 \
+  --bm25 --k1 0.82 --b 0.68
+```
+
+
+### Performing Reranking Using Pretrained Model
+
+
+Download pretrained IBM models:
+```bash
+wget https://rgw.cs.uwaterloo.ca/JIMMYLIN-bucket0/pyserini-models/ibm_model_1_bert_tok_20211117.tar.gz -P irst_test/
+tar -xzvf irst_test/ibm_model_1_bert_tok_20211117.tar.gz -C irst_test
+```
+
+Next we can run our script to get our reranking results.
+
+IRST (Sum) 
+```bash
+python -m pyserini/scripts/reranker_ibm_colbert.py 
```

Comment:
can this be re-packaged as `python -m pyserini.search.lucene.irst`?

Reply:
Yes, but the scripts are quite similar to python -m pyserini.search.lucene.tprob. Just tprob is an end-to-end and irst is a reranker.

Reply:
I end up adding a python -m pyserini.search.lucene.trob.rerank option.